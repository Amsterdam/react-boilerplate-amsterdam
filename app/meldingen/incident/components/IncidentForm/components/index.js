import DescriptionWithClassificationInput from './DescriptionWithClassificationInput/';
import DateTimeInput from './DateTimeInput/';
import FileInput from './FileInput/';
import HiddenInput from './HiddenInput/';
import TextInput from './TextInput/';

export default {
  DateTimeInput,
  DescriptionWithClassificationInput,
  FileInput,
  HiddenInput,
  TextInput
};
