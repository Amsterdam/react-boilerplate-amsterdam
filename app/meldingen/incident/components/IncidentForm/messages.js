/*
 * IncidentForm Messages
 *
 * This contains all the text for the IncidentForm component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.IncidentForm.header',
    defaultMessage: 'This is the IncidentForm component !',
  },
});
