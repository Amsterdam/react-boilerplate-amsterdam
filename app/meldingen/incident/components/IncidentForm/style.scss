.incident-form {
  input,
  textarea {
    background-color: white;
    padding: 2px 8px;
  }
}
