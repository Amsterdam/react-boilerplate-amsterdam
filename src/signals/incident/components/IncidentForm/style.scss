.incident-form {
  .field-errors {
    color: red;
  }

  .mode_input, .mode_upload {
    margin-top: 10px;
  }
}
