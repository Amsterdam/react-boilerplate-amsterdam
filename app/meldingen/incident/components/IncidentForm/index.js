/**
*
* IncidentForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { FormGenerator } from 'react-reactive-form';

// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

import './style.scss';

class IncidentForm extends React.Component {
  constructor(props) {
    super(props);

    this.setForm = this.setForm.bind(this);
    this.setValues = this.setValues.bind(this);
    this.handleReset = this.handleReset.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(props) {
    this.setValues(props.incident, true);
  }

  setForm(form, incident) {
    this.form = form;
    this.form.meta = {
      form: this.form,
      handleSubmit: this.handleSubmit,
      getClassification: this.props.getClassification,
      setIncident: this.props.setIncident
    };

    this.setValues(incident);
  }

  setValues(incident, onlyWatchedItems = false) {
    if (this.form && this.form.controls) {
      window.setTimeout(() => {
        Object.keys(this.form.controls).map((key) => {
          const control = this.form.controls[key];
          if (!onlyWatchedItems || (onlyWatchedItems && control.meta.watch)) {
            control.setValue(incident[key]);
          }
          return true;
        });
      });
    }
  }

  handleReset() {
    this.form.reset();
  }

  handleSubmit(e) {
    e.preventDefault();
  }

  render() {
    return (
      <div className="incident-form">
        <form onSubmit={this.handleSubmit}>
          <FormGenerator
            onMount={(form) => this.setForm(form, this.props.incident)}
            fieldConfig={this.props.fieldConfig}
          />
        </form>
      </div>
    );
  }
}

IncidentForm.propTypes = {
  fieldConfig: PropTypes.object.isRequired,
  incident: PropTypes.object.isRequired,
  getClassification: PropTypes.func.isRequired,
  setIncident: PropTypes.func.isRequired
};

export default IncidentForm;
