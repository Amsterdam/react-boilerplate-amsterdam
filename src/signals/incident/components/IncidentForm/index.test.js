import React from 'react';
import { shallow } from 'enzyme';

import IncidentForm from './index';

describe('<IncidentForm />', () => {
  let props;

  beforeEach(() => {
    props = {
      fieldConfig: {
        controls: {}
      },
      incidentContainer: {
        incident: {}
      },
      wizard: {

      },
      getClassification: jest.fn(),
      updateIncident: jest.fn(),
      createIncident: jest.fn(),
      isAuthenticated: false
    };
  });

  it('expect to render correctly', () => {
    const wrapper = shallow(
      <IncidentForm {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
