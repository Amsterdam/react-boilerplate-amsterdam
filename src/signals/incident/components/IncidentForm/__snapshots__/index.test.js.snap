// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentForm /> expect to render correctly 1`] = `
<div
  className="incident-form"
>
  <form
    onSubmit={[Function]}
  >
    <FormGenerator
      fieldConfig={
        Object {
          "controls": Object {},
        }
      }
      onMount={[Function]}
      onUnmount={[Function]}
    />
  </form>
</div>
`;
